docs/tech_specs/0004_store_screen_responses.md

```diff
@@ -0,0 +1,73 @@
+
+**What kind of load does this put on our S3 resources?**
+
+A few important variables: we foresee wanting to retain no more than 90 days of history. And from trying out a couple example files, the blobs should be around 5-5.5KB in size. 
+
+Splunk currently reports 90 active screen urls. (Some of the urls are reused on multiple units. For example, Ashmont has 2 Pre-Fare sets, and each set has 2 screen instances of our app url.) Splunk reports about 21.6k data requests an hour. If we catalog the json for each response at this scale, we'd have 518k files posted to S3 across 90 S3 prefixes. (Which is 46 million files stored over 90 days, or 244GB)
+
+In the short term, we plan to add 200 new screens: about 180 e-ink and 20 miscellaneous screens by the end of the year. E-inks average 200 requests/hr while all other screens average around 240 requests/hr. This raises estimated data requests/hr to 46.4k.
+
+In the long term, we may have a total of 1000 screens (countdown signs, 125 pre-fare pairs). This raises estimated data requests/hr to 214.4k. (463 million files over 90 days, 2.4TB)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Cost. How much does this go against our S3 budget? Preliminary budget calculation (with help from Paul S): this would be a 1% increase of our AWS bill. If this poses a problem, we can explore reducing retention, reducing duplicate posts for screens that share a url, and perhaps reducing frequency of posting altogether.
+- Limited querying. If we needed more query fields, then there are DB options that would be better. But it doesn't seem we'll really be using these logs to do precise querying at this time. It's more for triaging particular screen issues.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Considered alternatives:
+- Postgres with JSONB column
+- S3 in Parquet file format
+- Logging to Sentry using the json source_type
+
+We stuck with the S3 solution rather than introduce a DB because:
+- It should cover our requirements
+- We already use S3 in the project stack
+- The cost shouldn't be a problem (after discussing with some engineering leads)
+
+As for using Parquet (recommended by the LAMP team) that would just help make the data more queryable, which we don't really need for our use case.
+
+As for logging to Sentry, this was kind of a late-breaking idea, and I didn't immediately find anyone in CTD who has programmatically queried Sentry in this way before, so I just stuck with the S3 option as prime choice. (If the PR reviewers have other insight, this could be an interesting option.)
+
+We also considered these alternatives to streamline the frontend rendering before deciding it was out of scope for this work:
+- Store jpgs of the screen content
+- Store html of the screens (tricky, because some screens have frontend paging that will not be represented in the html.)
```

Comment:
The frontend paging issue may actually just be the same situation as backend paging, but instead of pages being omitted in the response, they are omitted from the DOM. We would be able to see all pages in the server response, but would get a more accurate snapshot if we were to look at the HTML. (Not really advocating for this, just something I thought about.)